addons/arsenal/functions/fnc_verifyLoadout.sqf

```diff
@@ -23,279 +23,51 @@ if (count _loadout == 2) then {
     _loadout = _loadout select 0;
 };
 
-private _cfgWeapons = configFile >> "CfgWeapons";
-private _cfgMagazines = configFile >> "CfgMagazines";
-private _cfgVehicles = configFile >> "CfgVehicles";
-private _cfgGlasses = configFile >> "CfgGlasses";
-
-private _weapons = GVAR(virtualItems) get IDX_VIRT_WEAPONS;
-private _attachments = GVAR(virtualItems) get IDX_VIRT_ATTACHMENTS;
-
 private _name = "";
-private _nullItemsAmount = 0;
-private _unavailableItemsAmount = 0;
+private _itemArray = [];
 private _nullItemsList = [];
 private _unavailableItemsList = [];
 
-// Search for all items and turn them into config case; Don't touch other value types
-private _fnc_toConfigCase = {
+// Search for all items and check their availability
+private _fnc_filterLoadout = {
     _this apply {
-        if (_x isEqualType "") then {
-            if (_x != "") then {
-                _name = _x call EFUNC(common,getConfigName);
+        if (_x isEqualType "" && {_x != ""}) then {
+            _name = _x call EFUNC(common,getConfigName);
 
-                // If item doesn't exist in config, "" is returned
-                // Just return unaltered item name in that case, so it can be documented as being unavailable
-                if (_name != "") then {
-                    _name
-                } else {
-                    _x
-                };
+            // If item doesn't exist in config, "" is returned
+            if (_name == "") then {
+                _nullItemsList pushBack _x;
             } else {
-                _x
+                // Check if item or its base weapon exist in the arsenal
+                if !(_name in GVAR(virtualItemsFlatAll)) then {
+                    _name = _name call FUNC(baseWeapon);
+                    if !(_name in GVAR(virtualItemsFlatAll)) then {
```

Comment:
Do we want `GVAR(virtualItemsFlatAll)` or `GVAR(virtualItemsFlat)`? I feel `GVAR(virtualItemsFlat)` is more appropriate, seeing as `GVAR(virtualItemsFlatAll)` includes unique items.

Reply:
I like the idea of `GVAR(virtualItemsFlatAll)` being more flexible, but it can be used to duplicate items, so `GVAR(virtualItemsFlat)` it is.

Reply:
```suggestion
                if !(_name in GVAR(virtualItemsFlat)) then {
                    _name = _name call FUNC(baseWeapon);
                    if !(_name in GVAR(virtualItemsFlat)) then {
```